dist/src/main/java/io/zeebe/broker/BrokerSpringServerCustomizer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker;
+
+import static java.net.InetAddress.getByName;
+
+import io.prometheus.client.CollectorRegistry;
+import io.zeebe.broker.system.configuration.BrokerCfg;
+import io.zeebe.util.exception.UncheckedExecutionException;
+import java.net.UnknownHostException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.autoconfigure.metrics.export.prometheus.PrometheusMetricsExportAutoConfiguration;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.web.server.WebServerFactoryCustomizer;
+import org.springframework.boot.web.servlet.server.ConfigurableServletWebServerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BrokerSpringServerCustomizer
+    implements WebServerFactoryCustomizer<ConfigurableServletWebServerFactory> {
+
+  @Autowired BrokerCfg brokerCfg;
+
+  @Override
+  public void customize(ConfigurableServletWebServerFactory server) {
+    final var networkCfg = brokerCfg.getNetwork();
+    // trigger application of defaults so that the monitoring config no longer has null values
+    networkCfg.applyDefaults();
```

Comment:
The broker has not initialized the config yet. Also, we cannot initialize the full `BrokerCfg` configuration yet, because we don't yet know the folder in which the broker will run. 

I don't think it is a big deal, because right now this is an idempotent operation. This branch of the config tree does not depend on anything the broker adds later. 

I'm not a big fan of it, but I think it's ok.

Reply:
Yeah, when I said fragile I meant because of this - right now it's an idempotent operation, but it's not really enforced, so I can imagine that changing and us spending time trying to figure out why this is behaving unexpectedly :sweat_smile: But you're probably right and I'm being too cautious here